src/libvirtApi/common.js

```diff
@@ -161,88 +161,91 @@ function doUsagePolling(name, connectionName, objPath) {
+            store.dispatch(updateVm(props));
+        }
+    } catch (ex) {
+        console.warn(`GetStats(${name}, ${connectionName}) failed: ${ex.toString()}`);
+    }
+    delayPolling(() => doUsagePolling(name, connectionName, objPath), null);
 }
 
-function getLoggedInUser() {
-    logDebug(`GET_LOGGED_IN_USER:`);
-    return cockpit.user().then(loggedUser => store.dispatch(setLoggedInUser({ loggedUser })));
+async function getLoggedInUser() {
+    const loggedUser = await cockpit.user();
+    logDebug(`GET_LOGGED_IN_USER:`, loggedUser);
+    store.dispatch(setLoggedInUser({ loggedUser }));
 }
 
-export function getLibvirtVersion({ connectionName }) {
-    return call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" })
-            .then(version => store.dispatch(updateLibvirtVersion({ libvirtVersion: version[0].v })));
+export async function getLibvirtVersion({ connectionName }) {
+    const [version] = await call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" });
+    store.dispatch(updateLibvirtVersion({ libvirtVersion: version.v }));
 }
 
-function getNodeMaxMemory({ connectionName }) {
+async function getNodeMaxMemory({ connectionName }) {
     // Some nodes don"t return all memory in just one cell.
     // Using -1 == VIR_NODE_MEMORY_STATS_ALL_CELLS will return memory across all cells
-    return call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" })
-            .then(stats => store.dispatch(setNodeMaxMemory({ memory: stats[0].total })))
-            .catch(ex => {
-                console.warn("NodeGetMemoryStats failed:", ex.toString());
-                return Promise.reject(ex);
-            });
+    try {
+        const [stats] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" });
+        store.dispatch(setNodeMaxMemory({ memory: stats.total }));
+    } catch (ex) {
+        console.warn("NodeGetMemoryStats failed:", ex.toString());
+        throw ex;
+    }
 }
 
-function getOsInfoList () {
+async function getOsInfoList () {
     logDebug(`GET_OS_INFO_LIST():`);
-    return python.spawn(getOSListScript, null, { err: "message", environ: ['LC_ALL=C.UTF-8'] })
-            .then(osList => {
-                parseOsInfoList(osList);
-            })
-            .catch(ex => {
-                console.error(`get os list returned error: "${JSON.stringify(ex)}"`);
-                parseOsInfoList('[]');
-            });
+    try {
+        const osList = await python.spawn(getOSListScript, null, { err: "message", environ: ['LC_ALL=C.UTF-8'] });
+        parseOsInfoList(osList);
+    } catch (ex) {
+        console.error(`get os list returned error: "${JSON.stringify(ex)}"`);
+        parseOsInfoList('[]');
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-075343-fbf2cd7c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)